### PR TITLE
NULL dereference fixes

### PR DESCRIPTION
**Pull request by bmiklautz**

### CODE REVIEW

File: winpr/libwinpr/utils/wlog/wlog.c [hunk 1]
```diff
@@ -353,7 +353,8 @@ int WLog_ParseFilter(wLogFilter* filter, LPCSTR name)
 
 	while ((p = strchr(p, '.')) != NULL)
 	{
-		filter->Names[count++] = p + 1;
+		if (count < filter->NameCount)
```
Comment (nfedera) [inline winpr/libwinpr/utils/wlog/wlog.c]: IMHO this is always true in this context - no ?

File: winpr/libwinpr/utils/wlog/wlog.c [hunk 2]
```diff
@@ -332,7 +332,7 @@ int WLog_ParseFilter(wLogFilter* filter, LPCSTR name)
 
 	names = _strdup(name);
 	filter->NameCount = count;
-	filter->Names = (LPSTR*) malloc(sizeof(LPSTR) * (count + 1));
+	filter->Names = (LPSTR*) calloc((count + 1UL), sizeof(LPSTR));
```
Comment (nfedera) [inline winpr/libwinpr/utils/wlog/wlog.c]: name (the parameter) and names (result of strdup) both must be checked for NULL

File: winpr/libwinpr/utils/wlog/wlog.c [hunk 3]
```diff
@@ -405,7 +407,8 @@ int WLog_ParseFilters()
 
 	while ((p = strchr(p, ',')) != NULL)
 	{
-		strs[count++] = p + 1;
+		if (count < g_FilterCount)
```
Comment (nfedera) [inline winpr/libwinpr/utils/wlog/wlog.c]: same question: can this really be false in this context ?

File: winpr/libwinpr/utils/wlog/wlog.c [hunk 4]
```diff
@@ -508,7 +512,8 @@ int WLog_ParseName(wLog* log, LPCSTR name)
 
 	while ((p = strchr(p, '.')) != NULL)
 	{
-		log->Names[count++] = p + 1;
+		if (count < log->NameCount)
```
Comment (nfedera) [inline winpr/libwinpr/utils/wlog/wlog.c]: same question: how can this be false in this context ?

**Comment by nfedera** [comment]
Nice. +1 after the WLog_ParseFilter name param and strdup is NULL checked

**Comment by akallabeth** [comment]
+1 for the additional fixes

**Pull request merged**